destination/writer/writer.go

```diff
@@ -0,0 +1,286 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/conduitio-labs/conduit-connector-redshift/columntypes"
+	"github.com/conduitio-labs/conduit-connector-redshift/config"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/huandu/go-sqlbuilder"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	// metadataFieldTable is a name of a record metadata field that stores a Redshift table name.
+	metadataFieldTable = "redshift.table"
+	// keySearchPath is a key of get parameter of a datatable's schema name.
+	keySearchPath = "search_path"
+)
+
+// Writer implements a writer logic for Redshift destination.
+type Writer struct {
+	db          *sqlx.DB
+	table       string
+	keyColumns  []string
+	columnTypes map[string]string
+}
+
+// NewWriter creates new instance of the Writer.
+func NewWriter(ctx context.Context, driverName string, config config.Destination) (*Writer, error) {
+	var err error
+
+	writer := &Writer{
+		table:      config.Table,
+		keyColumns: config.KeyColumns,
+	}
+
+	writer.db, err = sqlx.Open(driverName, config.DSN)
+	if err != nil {
+		return nil, fmt.Errorf("open db connection: %w", err)
+	}
+
+	u, err := url.Parse(config.DSN)
```

Comment:
Thoughts out loud: wouldn't it make sense to make the `config.DSN` field `*url.URL`?

Reply:
Do you mean parse `config.DSN` and call `String()` method to get `config.DSN`?

Reply:
https://github.com/conduitio-labs/conduit-connector-redshift/blob/4563a30da5181f2fd18b3a7b7e6205229749444c/config/configuration.go#L34
Yeah, I thought about changing the `Configuration.DSN`'s type to `*url.URL` and parsing it right in the `parseCommon` function. This way we can avoid parsing this field in Source and Destination.

Reply:
Hm. It could be a good idea 👍 

I think I will do it, but definitely in a different pull request. Thank you.